frontend/src/features/properties/components/GeocoderAutoComplete.tsx

```diff
@@ -52,30 +51,32 @@ export const GeocoderAutoComplete: React.FC<IGeocoderAutoCompleteProps> = ({
   const { handleBlur } = useFormikContext<any>();
   const errorTooltip = error && touch && displayErrorTooltips ? error : undefined;
 
-  const search = useCallback(
-    (val: string, abort: boolean) =>
-      debounce(async (val: string, abort: boolean) => {
+  const debouncedSearch = React.useRef(
```

Comment:
The previous format of this debounce function was not working.

Reply:
Can you explain why?  To me it looks like you're now calling the debounce function immediately rather than as a function.

Reply:
Need to look into why the previous version wasn't working, I was under the impression that pattern worked.
This pattern stores the debounced function in a global ref that isn't updated after every render. (note that calling debounce()) on a function doesn't execute that function, it just wraps that function and returns the wrapped function). All subsequent calls to debouncedSearch are guaranteed to execute against the same debounced wrapper, which ensures that the wrapper has a chance to debounce all of the calls appropriately.